src/main/kotlin/edu/byu/mealplanningassistant/repository/RecipesRepository.kt

```diff
@@ -53,16 +54,29 @@ class RecipesRepository{
         }
     }
 
-    fun getRecipesWithTags(tagSet: HashSet<String>) = getFromDB(`all`("tags", tagSet))
+    fun getRecipesWithTags(tagSet: HashSet<String>, sort : Boolean = false) = getFromDB(`all`("tags", tagSet), sort)
+
+    fun getRecipesWithTagsAndIngredients(tagSet: HashSet<String>, ingredientSet: HashSet<String>, sort : Boolean = false) =
+        getFromDB(and(`all`("tags", tagSet), `all`("ingredients", ingredientSet)), sort)
+
 
     // TODO I'm not sure if this is supposed to do something different than getAllRecipes later... I don't think it's pertinent to decide this now, but should discuss and consolidate if possible later
-    fun getRecipes() = getFromDB()
+    fun getRecipes() = getFromDB(null, false)
 
-    private fun getFromDB(filter: Bson? = null) : List<Recipe>{
-        println("filter : $filter")
+    private fun getQueryResults(filter: Bson? = null, sort : Boolean) : Iterator<Document>{
         val collection = getDatabase().getCollection("recipes")
+        return if(sort){
+            (if (filter != null) collection.find(filter).sort(Sorts.descending("rating")).iterator() else collection.find().sort(Sorts.descending("rating")).iterator())
+        } else{
+            (if (filter != null) collection.find(filter).sort(Sorts.descending("rating")).iterator() else collection.find().sort(Sorts.descending("rating")).iterator())
+        }
+    }
```

Comment:
The two branches of `if(sort)` appear to be exactly the same, is it supposed to be that way?